### PR TITLE
Cache ToolkitWrapper.is_available/installed/licensed

### PR DESCRIPTION
**Pull request by mattwthompson**
This PR fixes at least one minor issue (#443, #485) and addresses a TODO.
* Appropriately raise `ToolkitUnavailableException` vs `LicenseError` based on if OpenEye is installed and/or licensed. I started just by clobbering the two, but I think the solution allows the two cases to separately be caught.
* Cache `ToolkitWrapper._is_available` to prevent unnecessary calls during other method calls

- [x] Run a code profiler to see if the second point above actually provides any speedup: provides an inconsequential speedup of 1000x in a fast function call https://gist.github.com/mattwthompson/2a1687d211f4877e593e48aa8bd8d37d
- [x] Tag issue being addressed
- [x] Add [tests](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openforcefield/tree/master/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

### CODE REVIEW

**Review by j-wags — APPROVED** [review]
Awesome. This looks great. There were some strange CI failures that went away upon rerunning. I've updated the releasenotes. 

Consider committing the `TODO` removal I suggested, and then merge when ready :-)

File: openforcefield/utils/toolkits.py [hunk 1]
```diff
@@ -585,16 +580,36 @@ def __init__(self):
...
+        for (tool, license_func) in cls._license_functions.items():
+            try:
+                module = importlib.import_module("openeye." + tool)
+            except (ImportError, ModuleNotFoundError):
+                return False
+            else:
+                all_licensed &= getattr(module, license_func)()
+        return all_licensed
+
+    # TODO: Add back in oetools arg? It doesn't seem used
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply openforcefield/utils/toolkits.py]: My concern here was only that it's removing an argument from a  public API point

[the remark this answers is not in the document]
Comment (j-wags) [reply openforcefield/utils/toolkits.py]: Oh, good point. I'll update the releasenotes to indicate an API-breaking change.

**Comment by j-wags** [comment]
Ok, I've documented the API-breaking change in the release history. This should be good to merge when you're ready :-)

**Comment by mattwthompson** [comment]
Thanks for the review and quick turnaround!

**Pull request merged**